### PR TITLE
add testplans for crosswordpuzzle

### PR DESCRIPTION
**Pull request by Aaron2000119**
Closes https://github.com/Gamify-IT/issues/issues/179

### CODE REVIEW

**Review by Gr33ndev — APPROVED** [review]

File: dev-manuals/test-plans/minigames/crosswordpuzzle-frontend/plans/u-crosswordpuzzle-2-start-game.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+A configuration has been stored in the backend and selected in the frontend.
+
+## Postcondition
+
+A crosswordpuzzle, the questions and a `Evaluate` button are visible.
+
+## Typical procedure
+
+1. Open the crosswordpuzzle
```
Comment (delvh) [inline dev-manuals/test-plans/minigames/crosswordpuzzle-frontend/plans/u-crosswordpuzzle-2-start-game.md:21]:
```suggestion
1. Open the crosswordpuzzle with the given configuration
```

File: dev-manuals/test-plans/minigames/crosswordpuzzle-frontend/plans/u-crosswordpuzzle-1-start-page.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+A configuration has been stored in the backend and selected in the frontend.
+
+## Postcondition
+
+The questions and a start button are visible.
+
+## Typical procedure
+
+1. Open the crossword-puzzle
```
[the remark this answers is not in the document]
Comment (delvh) [reply dev-manuals/test-plans/minigames/crosswordpuzzle-frontend/plans/u-crosswordpuzzle-1-start-page.md:21]: Also, something doesn't add up here:
why is in this test case the same setup and steps, but a different result than in `u.crosswordpuzzle-2` is achieved?
Please change the text to be more descriptive of the difference.

File: dev-manuals/test-plans/minigames/crosswordpuzzle-frontend/plans/u-crosswordpuzzle-3-finish-game.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+A modal is shown where you result is displayed.
+
+## Typical procedure
+
+1. Open the crossword-puzzle
+2. Press the `start` button
+3. Input the correct answers in the crosswordpuzzle
+4. Press the `Evaluate` button
+5. A modal is shown that says `Congratulations!` `Everything right!`
```
Comment (delvh) [inline dev-manuals/test-plans/minigames/crosswordpuzzle-frontend/plans/u-crosswordpuzzle-3-finish-game.md:25]: A modal? Not a toast?
```suggestion
5. A success message is shown
```

File: dev-manuals/test-plans/minigames/crosswordpuzzle-frontend/plans/u-crosswordpuzzle-3-finish-game.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+## Alternative procedures
+3.1 Input wrong answers in the crosswordpuzzle \
+    3.1.1 Press the `Evaluate` button \
+    3.1.2 A modal is shown that says `Not the correct answers` `Maybe the next time`
```
Comment (delvh) [inline dev-manuals/test-plans/minigames/crosswordpuzzle-frontend/plans/u-crosswordpuzzle-3-finish-game.md:30]:
```suggestion
    3.1.2 A failure message is shown
```

File: dev-manuals/test-plans/minigames/crosswordpuzzle-frontend/plans/u-crosswordpuzzle-4-restart-game.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+Author: Aaron Schmid
+
+
+## Description
+
+This use case verifys that the crosswordpuzzle can be restarted.
+
+## Precondition
+
+A configuration has been stored in the backend and selected in the frontend.
```
Comment (delvh) [inline dev-manuals/test-plans/minigames/crosswordpuzzle-frontend/plans/u-crosswordpuzzle-4-restart-game.md:13]: You can start after a successful round (prior to submitting the solution?) to omit 3/4 of the typical procedure.

File: dev-manuals/test-plans/minigames/crosswordpuzzle-frontend/plans/u-crosswordpuzzle-4-restart-game.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+1. Open the crossword-puzzle
+2. Press the `start` button
+3. Input the correct answers in the crosswordpuzzle
+4. Press the `Evaluate` button
+5. Press the `Back to welcome page` button
+6. The start page is shown again
+
+## Alternative procedures
+5.1 Press the `Close` button \
```
Comment (delvh) [inline dev-manuals/test-plans/minigames/crosswordpuzzle-frontend/plans/u-crosswordpuzzle-4-restart-game.md:29]:
```suggestion
## Alternative procedures

5.1 Press the `Close` button \
```

**Review by delvh — CHANGES_REQUESTED** [review]

File: dev-manuals/test-plans/minigames/crosswordpuzzle-frontend/plans/u-crosswordpuzzle-3-finish-game.md (hunk 3 above)
Comment (Aaron2000119) [reply dev-manuals/test-plans/minigames/crosswordpuzzle-frontend/plans/u-crosswordpuzzle-3-finish-game.md]: After evaluationg a modal with the restart and the close button shows up, where the result is displayed.

File: dev-manuals/test-plans/minigames/crosswordpuzzle-frontend/plans/u-crosswordpuzzle-1-start-page.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Aaron2000119) [reply dev-manuals/test-plans/minigames/crosswordpuzzle-frontend/plans/u-crosswordpuzzle-1-start-page.md]: The first test case is for the start page and the second one shows the page with the crosswordpuzzle.

[the remark this answers is not in the document]
Comment (delvh) [reply dev-manuals/test-plans/minigames/crosswordpuzzle-frontend/plans/u-crosswordpuzzle-1-start-page.md]: Yes. and you have both times have the same `1.`, but a different check in `2.`.
That's not possible unless we had a Schrödinger's Program.
That's what I mean: Why would the exact same set of actions lead to different assertions?
Hence, you should re-write `1.` for (at least) one testcase.

[the remark this answers is not in the document]
Comment (delvh) [reply dev-manuals/test-plans/minigames/crosswordpuzzle-frontend/plans/u-crosswordpuzzle-1-start-page.md]: My bad. Just noticed that I overlooked that the check is `3.` in the second one, not `2.`. It's fine then.

**Review by delvh — APPROVED** [review]

**Pull request merged**